droidlet/perception/semantic_parsing/nsp_transformer_model/caip_dataset.py

```diff
@@ -120,30 +120,20 @@ def __getitem__(self, idx):
             p_text, p_tree = t
         except ValueError as e:
             print(e)
-        text, tree = tokenize_linearize(
-            p_text, p_tree, self.tokenizer, self.full_tree, self.word_noise
-        )
+
+        text, tree = tokenize_text_tree(p_text, p_tree, self.tokenizer, self.word_noise)
 
         text_idx_ls = self.tokenizer.convert_tokens_to_ids(text.split())
-        tree_idx_ls = [
-            [
-                self.tree_idxs[w],
-                bi,
-                ei,
-                text_span_start,
-                text_span_end,
-                (
-                    self.tree_idxs[fixed_span_val]
-                    if type(fixed_span_val) == str
-                    else fixed_span_val
-                ),
-            ]
-            for w, bi, ei, text_span_start, text_span_end, fixed_span_val in [
-                ("<S>", -1, -1, -1, -1, -1)
-            ]
-            + tree
-            + [("</S>", -1, -1, -1, -1, -1)]
-        ]
+        tree_idx_ls = self.tokenizer.convert_tokens_to_ids(tree.split())
+
+        # Truncate tokens if it exceeds the maximal length
+        # which is 512 for bert tokenizer
+        max_length = 512
```

Comment:
this needs to come from a config or opt, and be much more visible.  As it is, it will silently truncate

Reply:
> this needs to come from a config or opt, and be much more visible. As it is, it will silently truncate

I tried to retrieve from tokenizer but couldn't find corresponding attribute. I will move it to opt.